django_ipam/base/models.py

```diff
@@ -55,6 +59,70 @@ def request_ip(self, options=None):
             return ip_address
         return None
 
+    def import_csv(self, file):
+        ipaddress_model = swapper.load_model("django_ipam", "IpAddress")
+        subnet_model = swapper.load_model("django_ipam", "Subnet")
+        if file.name.endswith(('.xls', '.xlsx')):
+            book = xlrd.open_workbook(file_contents=file.read())
+            sheet = book.sheet_by_index(0)
+            row = []
+            for row_num in range(sheet.nrows):
+                row.append(sheet.row_values(row_num))
+            reader = iter(row)
+        else:
+            reader = csv.reader(StringIO(file.read().decode('utf-8')),
+                                delimiter=',')
+        subnet_name = next(reader)[0].strip()
+        subnet_value = next(reader)[0].strip()
+        try:
+            subnet_exists = subnet_model.objects.filter(name=subnet_name, subnet=subnet_value).exists()
+        except (ValidationError, ValueError):
+            return False
```

Comment:
First concern: this approach is not very pythonic.

In Python it is common to raise exceptions when something unexpected happens and then catch that exception anywhere it needs to be caught to handle it.

Example:
```python
class CsvImportException(Exception):
    pass
```

Then you can raise it with:

```python
raise CsvImportException(_('explain what went wrong here'))
```

When you catch it you can do:

```python
try:
    perform_import()  # pseudo-code
except CsvImportException as e:
    # e.message will contain the information about what went wrong
    # you have to show this users
    print(e.message)
```

**Second concern**

This code looks like it can be simplified, what do you think of the following:

```python
try:
    subnet = subnet_model.objects.get(name=subnet_name,
                                      subnet=subnet_value)
except subnet_model.DoesNotExist:
    try:
        subnet = subnet_model(name=subnet_name,
                              subnet=subnet_value)
        subnet.full_clean()
        subnet.save()
    except ValidationError as e:
        # here django should give us a human
        # readable error message which we can pass
        # to our exception,
        # but please verify this before doing it this way
        # there may be a better way
        message = str(e)
        raise CsvImportException(message)
    except ValueError as e:
        # why are we getting a ValueError?
        message = _('what went wrong and how to fix it?')
        raise CsvImportException(message)
```